logistics_planning_base/models/logistics_schedule.py

```diff
@@ -158,10 +158,11 @@ def _compute_product_uom_qty(self):
     def _onchange_stock_move_id(self):
         self.ensure_one()
         if self.stock_move_id:
-            self.write({
-                "commitment_date": self.stock_move_id.date,
-                "schedule_finished": True,
-            })
+            self.write({"schedule_finished": True})
+            if self.type == 'output':
+                self.write({"commitment_date": self.stock_move_id.date})
+            else:
+                self.write({"effective_date": self.stock_move_id.date})
```

Comment:
I see it less effective than this implementation (only one `write` access, not tested):
```suggestion
            date_field = "commitment_date" if self.type == 'output' else "effective_date"
            self.write({
                date_field: self.stock_move_id.date,
                "schedule_finished": True,
            })            
```